### PR TITLE
DRILL-3583: Added a deprecated UserException#build method to avoid fu…

### PR DESCRIPTION
**Pull request by sudheeshkatkam**
…ture mishaps

@parthchandra please review.

### CODE REVIEW

File: common/src/main/java/org/apache/drill/common/exceptions/UserException.java [hunk 1]
```diff
@@ -534,6 +535,19 @@ public UserException build(final Logger logger) {
 
       return newException;
     }
+
+    /**
+     * builds a user exception or returns the wrapped one. If the error is a system error, the error message is logged
+     * to the given {@link Logger}.
```
Comment (adeneche) [inline common/src/main/java/org/apache/drill/common/exceptions/UserException.java]: you should probably update this comment

*the error message is logged to the UserException's logger*

**Comment by sudheeshkatkam** [comment]
Addressed Hakim's comments.

**Comment by adeneche** [comment]
LGTM +1 (non-binding)

**Comment by parthchandra** [comment]
+1

**Pull request closed without merging**